src/connectors/menu/connectMenu.js

```diff
@@ -114,165 +117,148 @@ export default function connectMenu(renderFn, unmountFn = noop) {
       );
     }
 
-    return {
-      $$type: 'ais.menu',
-
-      isShowingMore: false,
-
-      // Provide the same function to the `renderFn` so that way the user
-      // has to only bind it once when `isFirstRendering` for instance
-      toggleShowMore() {},
-      cachedToggleShowMore() {
-        this.toggleShowMore();
-      },
-
-      createToggleShowMore({ results, instantSearchInstance }) {
-        return () => {
-          this.isShowingMore = !this.isShowingMore;
-          this.render({ results, instantSearchInstance });
-        };
-      },
-
-      getLimit() {
-        return this.isShowingMore ? showMoreLimit : limit;
-      },
-
-      refine(helper) {
-        return facetValue => {
-          const [refinedItem] = helper.getHierarchicalFacetBreadcrumb(
-            attribute
+    return withShowMore(
+      ({ toggleShowMore, isShowingMore, onToggleShowMore }) => ({
```

Comment:
I'd rather have a way to share behaviour, which can influence every part of the widget, including search parameters. As @samouss said, the issue is broader than just here, and thus might not be the pattern that fits us best

Reply:
This is precisely what _I didn't want_ to do in this proposal – I tried another approach (see https://github.com/algolia/instantsearch.js/pull/4128#issuecomment-533633177). Not saying that one is better than another, just wanted to try something new and trigger discussion.

Here I wanted the connector to still be in control of everything with the given information (e.g. "is it showing more right now?"). I tried to avoid the black box architecture that React InstantSearch creates.